### DISCUSSION TITLE
Search queries based solely on Point ID

### DISCUSSION THREAD

**Discussion by PieReissyet** [qdrant/qdrant#1914] [Ideas]
Hello, I have been using Qdrant for the past month for a matching engine tool I am building.

I am using a dataflow pipeline to preprocess, embed and upload the embeddings to Qdrant for million of images, so far so good.

Then for a sub-selection of those images that depends on some parameters I have uploaded to Qdrant too using the payloads, I want to retrieve the closest matches.

In order to do that I am first scrolling across the results that match those parameters. Then I collect all the embeddings and for each and every one of them I am making a search query to retrieve the N closest matches.

Since all of the search request I am making are linked to embeddings already uploaded inside the Qdrant collection, isn't there any way to actually make search requests based solely on the id of the point so that I dont have to scroll across the whole collection to get the embeddings first ? 

I think this is not possible yet but if it does let me know.

Thanks for the hard work and the super useful tool !

**Comment by generall** [comment]
Hi @PieReissyet, have you tried https://qdrant.tech/documentation/search/#recommendation-api ?

**Comment by PieReissyet** [comment]
I was not sure how to use the negative and positive parameters but just read some more and it feels like using only one positive ID will fit my need, thanks a lot !
Is there any reason to not directly include that behaviour to the search API ?

**Reply by generall** [reply]
> Is there any reason to not directly include that behaviour to the search API ?

Those APIs have different schema, I didn't want to over-complicate it. Also we have plans on how to improve recommendation API mechanics on lower level

**Comment by Divya-Ravali** [comment]


**Reply by joein** [reply]
Hi @Divya-Ravali 
it's `with_vectors`, not `with_vector`

**Reply by joein** [reply]
Thanks, will look into it
